remote_attestation/java/src/com/google/oak/remote_attestation/Message.java

```diff
@@ -0,0 +1,356 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package com.google.oak.remote_attestation;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInput;
+import java.io.DataInputStream;
+import java.io.DataOutput;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import static java.lang.Math.toIntExact;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+/** Wrapper class for Remote Attestation Protocol messages. */
+public class Message {
+  static final int CLIENT_HELLO_HEADER = 1;
+  static final int SERVER_IDENTITY_HEADER = 2;
+  static final int CLIENT_IDENTITY_HEADER = 3;
+
+  /** Remote Attestation protocol version. */
+  public static final int PROTOCOL_VERSION = 1;
+  /**
+   * Size (in bytes) of the prefix that describes the size of a serialized array.
+   * Prefix is encoded with Little-Endian.
+   */
+  public static final int SERIALIZATION_PREFIX_SIZE_BYTES = 8;
```

Comment:
I'm not sure the name of this constant reflects what it actually does; serialization prefix of what?

Reply:
Renamed to `ARRAY_SERIALIZATION_PREFIX_LENGTH`